extension/src/standalone_pages/InstructorDashboardRenderVideoInfo.js

```diff
@@ -194,6 +220,50 @@ function renderPassiveFeedback(video, response) {
             }
         }
     });
+
+    let keywordsMap = new Map();
+
+    labels.forEach((timestamp) => {
+        getVideoContextKeyWords(keywordsMap, video, timestamp);
+    });
+
+    let canvas = document.getElementById(`passiveFeedback_${video.videoID}`);
+    canvas.onclick = function(evt) {
+        let firstPoint = chart.getElementAtEvent(evt)[0];
+
+        if (firstPoint) {
+            let label = chart.data.labels[firstPoint._index];
+            console.debug(label, keywordsMap[label]);
+            displayContextKeyWords(label, keywordsMap);
+        }
+    };
+}
+
+async function getVideoContextKeyWords(keywordsMap, video, timestamp) {
+    keywordsMap[timestamp] = false;
+    chrome.runtime.sendMessage({
+        type: "FetchVideoContextKeyWords",
+        videoID: video.videoID,
+        timestamp: timestamp,
+        duration: video.video_duration
+    }, (response) => {
+        keywordsMap[timestamp] = processContextKeyWords(response);
+    });
+
+}
+
+function processContextKeyWords(result) {
```

Comment:
ah, looks like there is a minor misunderstanding in the API call.

The keyword extraction scripts actually returns keywords/keyphrases. Each element in the array is a keyword/keyphrase. Thus, we don't need to do any merging and de-duplication.

I highly recommend that you render the keywords/keyphrases in a visually appealing way. A good way to achieve this with minimal effort is by using [badge](https://getbootstrap.com/docs/4.0/components/badge/). It would be great if badges from `bert` and `rake` are differentiated by color (e.g. `bert` badges are red and `rake` badges are blue). Please remember to insert margin as needed to avoid all badge stick together.